### PR TITLE
Converted to AutoPlugin.

### LINKED ISSUE #24: Convert to `AutoPlugin`?
Happy to do the work if you're agreed this is a good goal. Any links plugin migration guides appreciated.

Need to get #22 fixed first.

### PR DESCRIPTION
**Pull request by metasim**
Fixed #24.

Notes:
* I suggest merging #30 first.
* I'd appreciate particular attention to [PomReaderPlugin line 9](https://github.com/metasim/sbt-pom-reader/blob/metasim-fix-24/src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala#L9), to confirm that this is an acceptable way to provide for backward compatibility.
* README.md needs to be updated upon plugin release to indicate non-AutoPlugin use for older versions of sbt.

### CODE REVIEW

**Comment by jsuereth** [comment]
Can you rebase this so it only shows autoplugin related changes?

**Comment by metasim** [comment]
Will do.

File: src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.typesafe.sbt.pom
+
+import sbt._
+import sbt.plugins.JvmPlugin
+
+/** Plugin definition for Maven POM reader. */
+object PomReaderPlugin extends AutoPlugin {
+  object autoImport extends SbtPomKeys {
+    @deprecated("This is an AutoPlugin; settings are automatically added. Using this value is no longer required", "1.1")
+    lazy val useMavenPom = projectSettings
```
Comment (metasim) [inline src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala]: @jsuereth For backward compatibility with pre-AutoPlugin usages, is this the right way communicating to users that a initialization call is no longer needed?

**Comment by metasim** [comment]
@jsuereth Rebased.

**Comment by metasim** [comment]
Didn't mean to close this PR... not sure what happened. Sorry about that!

File: src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala (hunk 1 above)
Comment (jsuereth) [inline src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala]: Exactly!

File: build.sbt [hunk 2]
```diff
@@ -2,32 +2,22 @@ name := "sbt-pom-reader"
...
 
-sbtPlugin := true
+licenses += ("Apache-2.0", url("http://opensource.org/licenses/Apache-2.0"))
 
-publishMavenStyle := false
+sbtPlugin := true
 
 libraryDependencies ++= Dependencies.pluginDependencies
 
-git.baseVersion := "1.0"
```
Comment (jsuereth) [inline build.sbt]: Did we bump the version to 2.0?

File: src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  }
+  override def requires = JvmPlugin
+  override def trigger = allRequirements
```
Comment (jsuereth) [inline src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala]: I think this should actually not be "allRequirements", but we should make users explciitly enable it....

I'm on the fence.  What's your inclination here?

**Comment by jsuereth** [comment]
Generally LGTM.  A lot of nice work there!  

A few quesitons/follow up, then good to merge :)

File: build.sbt (hunk 2 above)
Comment (metasim) [inline build.sbt]: Wasn't sure. I'll bump it.

File: src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala (hunk 3 above)
Comment (metasim) [inline src/main/scala/com/typesafe/sbt/pom/PomReaderPlugin.scala]: I waffled on this one. In the end, I leaned toward less work for the users, given that in most cases with this plugin, the `pom.xml` file is the authoritative build definition, and the user is likely to just want to get something that loads into sbt with minimal ceremony. That said, it's your call :-)

**Comment by metasim** [comment]
@jsuereth Thanks!

File: build.sbt (hunk 2 above)
Comment (metasim) [inline build.sbt]: @jsuereth Just remembered I'd moved the line into `version.sbt` where it was set to `1.1`.

**Comment by metasim** [comment]
Created a more concise PR to replace this one.

**Pull request closed without merging**